src/ai/ai_help_structs.cc

```diff
@@ -1003,19 +1005,22 @@ void PlayersStrengths::add(Widelands::PlayerNumber pn,
                            uint32_t cs,
                            uint32_t land,
                            uint32_t oland,
-                           uint32_t o60l) {
+                           uint32_t o60l,
+                           int32_t ds) {
 	if (all_stats.count(opn) == 0) {
 		this_player_number = pn;
 		this_player_team = mytn;
-		all_stats.insert(std::make_pair(opn, PlayerStat(pltn, pp, op, o60p, cs, land, oland, o60l)));
+		all_stats.insert(
+		   std::make_pair(opn, PlayerStat(pltn, pp, op, o60p, cs, land, oland, o60l, ds)));
 	} else {
 		all_stats[opn].players_power = pp;
 		all_stats[opn].old_players_power = op;
 		all_stats[opn].old60_players_power = o60p;
 		all_stats[opn].players_casualities = cs;
 		all_stats[opn].players_land = land;
 		all_stats[opn].old_players_land = oland;
-		all_stats[opn].old60_players_land = oland;
+		all_stats[opn].old60_players_land = o60l;
```

Comment:
Shouldn't buildings get the same treatment?

Reply:
actually these stats are needed in the AI warfare code, they are not invented for the diplomacy.     
I am not sure whether we need to consider development of buildings over time for AI as I think they are less important and actual buildings should be sufficient. 
as they are only needed for the diploscore we could have them but we could easily keep them internal.